File: transaction-api/src/main/java/io/github/mathter/memifydb/transaction/xa/XaException.java
package io.github.mathter.memifydb.transaction.xa;

import javax.transaction.xa.XAException;

public class XaException extends XAException {
    public XaException() {
    }

    public XaException(String message, int errcode) {
        super(message);
        this.errorCode = errcode;
    }
}

File: transaction-api/src/main/scala/io/github/mathter/memifydb/transaction/xa/XaException.scala
package io.github.mathter.memifydb.transaction.xa

import javax.transaction.xa.XAException

class XaException private(message: String) extends XAException(message) {
  def this(message: String, errorCode: Int) = {
    this(message)
    this.errorCode = errorCode
  }
}